pkg/scheduler/preemption/preemption.go

```diff
@@ -333,10 +334,35 @@ func workloadFits(wlReq cache.FlavorResourceQuantities, cq *cache.ClusterQueue,
 				cohortResRequestable = cq.Cohort.RequestableResources[flvQuotas.Name]
 			}
 			for rName, rReq := range flvReq {
-				limit := flvQuotas.Resources[rName].Nominal
+				nominal := flvQuotas.Resources[rName].Nominal
+				// lendingLimit here represents how much quota can be lent from the cohort
+				var lendingLimit, borrowingLimit, limit int64
+				flagBorrow, flagLend := false, false
 				if flvQuotas.Resources[rName].BorrowingLimit != nil && allowBorrowing {
-					limit += *flvQuotas.Resources[rName].BorrowingLimit
+					flagBorrow = true
+					borrowingLimit = *flvQuotas.Resources[rName].BorrowingLimit
 				}
+				if features.Enabled(features.LendingLimit) && cq.Cohort != nil && allowBorrowing {
```

Comment:
Can you move this under L369, since this only happens under that condition, which makes the code more readable.